codec-http/src/test/java/io/netty/handler/codec/http/HttpClientCodecTest.java

```diff
@@ -282,4 +282,34 @@ int getReceivedCount() {
             return receivedCount;
         }
     }
+
+    @Test
+    public void testDecodesFinalResponseAfterSwitchingProtocols() {
+        String SWITCHING_PROTOCOLS_RESPONSE = "HTTP/1.1 101 Switching Protocols\r\n" +
+                "Connection: Upgrade\r\n" +
+                "Upgrade: TLS/1.2, HTTP/1.1\r\n\r\n";
+
+        HttpClientCodec codec = new HttpClientCodec(4096, 8192, 8192, true);
+        EmbeddedChannel ch = new EmbeddedChannel(codec, new HttpObjectAggregator(1024));
+
+        HttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "http://localhost/");
+        request.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE);
+        request.headers().set(HttpHeaderNames.UPGRADE, "TLS/1.2");
+        assertTrue("Channel outbound write failed.", ch.writeOutbound(request));
+
+        assertTrue("Channel inbound write failed.",
+                ch.writeInbound(Unpooled.copiedBuffer(SWITCHING_PROTOCOLS_RESPONSE, CharsetUtil.ISO_8859_1)));
+        Object switchingProtocolsResponse = ch.readInbound();
```

Comment:
just do:

```java
FullHttpResponse switchingProtocolsResponse = ch.readInbound();
```

Reply:
Well, but wouldn't it be a bit too implicit and hide the real purpose of the test? This test is basically checking if we do get back a decoded response and not just a plain buffer, like it was before the patch.

Reply:
Honestly I am not sure why this would be any different then the assertThat that would fail below. Both will fail because the type is not what you expect. So I prefer what I suggested as then you can just remove the assertThas and also the cast later on when releasing

Reply:
Both would fail, but with different message, and in the second case - this would be the assertion failing and not just some code. An assertion failing is a strong indicator to me that the code breaks something this test was really testing for, and not just an accidental breaking of the test caused by e.g. refactoring. I just want to avoid a situation that in the future this call returns an object of a different type, and someone just fixes the test by changing the type in the test instead of thinking why the test was like that. 

But anyway, I can change it if you insist. :)

Reply:
I am not super strong on it... if you are more  happy with your way then its fine.

Reply:
Ok thanks, so I leave it as it is.